skyportal/models/observation_plan.py

```diff
@@ -0,0 +1,347 @@
+        doc="The target GcnEvent.",
+    )
+    gcnevent_id = sa.Column(
+        sa.ForeignKey('gcnevents.id', ondelete='CASCADE'),
+        nullable=False,
+        index=True,
+        doc="ID of the target GcnEvent.",
+    )
+
+    localization = relationship(
+        'Localization',
+        back_populates='observationplan_requests',
+        doc="The target Localization.",
+    )
+    localization_id = sa.Column(
+        sa.ForeignKey('localizations.id', ondelete='CASCADE'),
+        nullable=False,
+        index=True,
+        doc="ID of the target Localization.",
+    )
+
+    payload = sa.Column(
+        psql.JSONB, nullable=False, doc="Content of the observation plan request."
+    )
+
+    status = sa.Column(
+        sa.String(),
+        nullable=False,
+        default="pending submission",
+        index=True,
+        doc="The status of the request.",
+    )
+
+    allocation_id = sa.Column(
+        sa.ForeignKey('allocations.id', ondelete='CASCADE'), nullable=False, index=True
+    )
+    allocation = relationship('Allocation', back_populates='observation_plans')
+
+    observation_plans = relationship(
+        'EventObservationPlan',
+        passive_deletes=True,
+        doc='Observation plans associated with this request.',
+    )
+
+    target_groups = relationship(
+        'Group',
+        secondary='observationplan_groups',
+        passive_deletes=True,
+        doc='Groups to share the resulting data from this request with.',
+    )
+
+    @property
+    def instrument(self):
+        return self.allocation.instrument
+
+
+ObservationPlanRequestTargetGroup = join_model(
+    'observationplan_groups', ObservationPlanRequest, Group
+)
+ObservationPlanRequestTargetGroup.create = (
+    ObservationPlanRequestTargetGroup.update
+) = ObservationPlanRequestTargetGroup.delete = (
+    AccessibleIfUserMatches('followuprequest.requester')
+    & ObservationPlanRequestTargetGroup.read
+)
+
+
+class EventObservationPlan(Base):
+    """Tiling information, including the event time, localization ID, tile IDs,
+    and plan name"""
+
+    observation_plan_request_id = sa.Column(
+        sa.ForeignKey('observationplanrequests.id', ondelete="CASCADE"),
+        nullable=False,
+        doc='ObservationPlanRequest ID',
+    )
+
+    observation_plan_request = relationship(
+        "ObservationPlanRequest",
+        foreign_keys=observation_plan_request_id,
+        doc="The request that this observation plan belongs to",
+    )
+
+    instrument_id = sa.Column(
+        sa.ForeignKey('instruments.id', ondelete="CASCADE"),
+        nullable=False,
+        doc='Instrument ID',
+    )
+
+    instrument = relationship(
+        "Instrument",
+        foreign_keys=instrument_id,
+        doc="The Instrument that this observation plan belongs to",
+    )
+
+    dateobs = sa.Column(
+        sa.ForeignKey('gcnevents.dateobs', ondelete="CASCADE"),
+        nullable=False,
+        index=True,
+        doc='UTC event timestamp',
```

Comment:
Emphasize that this is the time of the GCN trigger, not of the observations.

Reply:
extended this